README.md

```diff
@@ -1,2 +1,108 @@
-# promiscuous-mysql
-Promise-based MySQL client
+# Promiscuous MySQL [![Build Status](https://travis-ci.org/clipper-digital/promiscuous-mysql.svg?branch=master)](https://travis-ci.org/clipper-digital/promiscuous-mysql)
+
+Promisucous MySQL is a promise-based MySQL client built on top of the [mysql](https://www.npmjs.com/package/mysql) module.
+
+
+
+
+
+## Example
+
+```js
+const database = require('promiscuous-mysql')
+
+database.withConnection({
+  host: 'example.org',
+  user: 'bob',
+  password: 'secret',
+  database: 'my-database'
+}, (connection) => {
+  return connection.query('SELECT * FROM `items`')
+})
+  .then((result) => {
+    console.log(result)
+  })
+```
+
+
+
+
+## API - Usage
+
+### `database.createConnection(config) Returns: Connection`
+
+Creates a new `Connection` instance.
+
+* `config` (Object): The connection configuration. See the [`mysql` connection options](https://www.npmjs.com/package/mysql#connection-options) for details.
+
+
+
+### `database.withConnection(config, callback) Returns: Promise`
+
+Creates a new `Connection` instance, invokes the callback with the new instance, then ends the connection.
+
+* `config` (Object): The connection configuration. See the [`mysql` connection options](https://www.npmjs.com/package/mysql#connection-options) for details.
+* `callback` (Function: `function(connection) Returns: Promise`): A callback used to perform queries with the created connection. Must return a promise indicating whether the operation was successful. The value of the promise returned from the callback will become the return value for `withConnection()`. If there is an error creating the connection, the callback will not be invoked and the returned promise will be rejected.
+
```

Comment:
What is the purpose of this exactly why the callback why not just have the promise resolve with a connection and then use `.then` instead of a callback? is it just to have it auto close the connection after?